### ISSUE TITLE
tModLoader server fails to start on ARM64/aarch64 Linux: MonoMod "method or operation is not implemented"

### ISSUE THREAD

**Issue by Koooper** [bluemagic123/tModLoader#3793]
### Version

Latest 1.4.4-stable

### OS

Linux

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

When I launch tML server via the dedicated server management script (flags: -g), the server fails to start due to an exception in the library MonoMod.

### Log File

[server.log](https://github.com/tModLoader/tModLoader/files/12533477/server.log)

### Steps to reproduce

1. Install tML server on an ARM Linux server (in this case an Oracle Cloud Ampere VM) using the -g flag (steamcmd incompatibility)
2. Attempt to run the server (./manage-tModLoaderServer.sh start -g)

### Expected Behavior

The server starts as expected

### Actual Behavior

Fatal exception

### Reproduction frequency

Every time

### how fix
1. wait for monomod to update 

### Additional Information

Version:  Linux version 5.15.0-83-generic (buildd@bos02-arm64-001) (gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0, GNU ld (GNU Binutils for Ubuntu) 2.38)

**Comment by ghost** [comment]
i think that could also be related to my issue #3766

**Comment by Koooper** [comment]
FWIW, here's [the code snippet](https://github.com/MonoMod/MonoMod/blob/reorganize/src/MonoMod.Core/Platforms/Systems/LinuxSystem.cs) for the upstream part of MonoMod.

**Comment by kiritantakechi** [comment]
MonoMod has recently merged support for arm64 (see MonoMod/MonoMod#241).

**Comment by Koooper** [comment]
wow, did not expect any update to this issue. would love to try this out to close the issue, but unfortunately i don't have present access to any ARM systems. bump?

**Comment by otomay** [comment]
Uppp

Just building with new MonoMod version is enough?

**Comment by kiritantakechi** [comment]
> Uppp
> 
> Just building with new MonoMod version is enough?

There may be some trouble because Terraria doesn't support native arm yet, and FNA has recently migrated to SDL3, so simply replacing it via libraries doesn't seem to be able to get Terraria to support native arm anymore.

**Comment by otomay** [comment]
> > Uppp
> > Just building with new MonoMod version is enough?
> 
> There may be some trouble because Terraria doesn't support native arm yet, and FNA has recently migrated to SDL3, so simply replacing it via libraries doesn't seem to be able to get Terraria to support native arm anymore.

I run a terraria 1.4.4.9 vanilla server in an arm64 server without emulation, is this what ur talking about or am I missing something?

**Comment by kiritantakechi** [comment]
> > > Uppp
> > > Just building with new MonoMod version is enough?
> > 
> > 
> > There may be some trouble because Terraria doesn't support native arm yet, and FNA has recently migrated to SDL3, so simply replacing it via libraries doesn't seem to be able to get Terraria to support native arm anymore.
> 
> I run a terraria 1.4.4.9 vanilla server in an arm64 server without emulation, is this what ur talking about or am I missing something?

You're right. The TerrariaServer should be much easier to run natively since its dependencies are far simpler than the full game client's.

My main focus has been on testing the Terraria client, specifically on my Mac. FNA used to provide a script that made it very easy to compile the necessary native libraries for macOS. However, as FNA has been updated, that script is no longer suitable for the older, SDL2-based version of FNA that we might need.

**Comment by Chicken-Bones** [comment]
We will be sure to close this issue when ARM support is achieved. In the mean time you can track progress here https://discord.com/channels/103110554649894912/1405829919187144765

**Comment by Chicken-Bones** [comment]
This was completed (See links https://github.com/tModLoader/tModLoader/pull/2639). ARM Linux servers are fully supported, though `steamcmd` still requires some Docker workarounds that we are in the process of documenting
